### PR TITLE
Don't fail when yaml is invalid

### PR DESCRIPTION
**Pull request by petey**
Converts config-parser to promise interface.
When a parse error occurs, the parser will return a valid config that spits out the error message, and exits with a failure.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,45 +2,50 @@
...
+        .then(doc => ({
             jobs: Hoek.reach(doc, 'jobs'),
             workflow: Hoek.reach(doc, 'workflow')
-        })
-    ], callback);
+        }))
+        .catch(err => ({
+            jobs: {
+                main: {
+                    image: 'node:4',
```
Comment (d2lam) [inline index.js]: Why `node:4`?

File: index.js [hunk 2]
```diff
@@ -2,45 +2,50 @@
...
+                    steps: [{
```
Comment (d2lam) [inline index.js]: shouldn't this be `commands`

**Review by d2lam — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (petey) [reply index.js]: I dunno. I'll change it to `alpine`

File: index.js (hunk 2 above)
Comment (petey) [reply index.js]: right you are then.

File: index.js [hunk 3]
```diff
@@ -2,45 +2,52 @@
...
 const Yaml = require('js-yaml');
 const Hoek = require('hoek');
-const Async = require('async');
 
 const phaseValidateStructure = require('./lib/phase/structural');
 const phaseFlatten = require('./lib/phase/flatten');
 const phaseValidateFunctionality = require('./lib/phase/functional');
 const phaseGeneratePermutations = require('./lib/phase/permutation');
 
+const parseYaml = yaml => (new Promise((resolve) => {
```
Comment (stjohnjohnson) [inline index.js]: Can you make this one-line?

**Review by minzcmu — APPROVED** [review]
👍 
fellcat

**Review by stjohnjohnson — APPROVED** [review]
Looking good! 👍

**Review by d2lam — APPROVED** [review]

**Review by FenrirUnbound — APPROVED** [review]
👍 

Looks good to me

345t9873154

File: index.js (hunk 3 above)
Comment (petey) [reply index.js]: fixed.

**Pull request merged**